presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestSphericalGeoFunctions.java

```diff
@@ -203,4 +203,63 @@ private void assertArea(String wkt, double expectedArea)
     {
         assertFunction(format("ABS(ROUND((ST_Area(to_spherical_geography(ST_GeometryFromText('%s'))) / %f - 1 ) * %d, 0))", wkt, expectedArea, 10000), DOUBLE, 0.0);
     }
+
+    @Test
+    public void testLength()
+    {
+        double distance1 = 1365691.096620299;
+        double distance2 = 2985175.539584118;
+        double length = distance1 + distance2;
+
+        // Empty linestring returns null
+        assertLength("LINESTRING EMPTY", null);
+
+        // Linestring with one point has length 0
+        assertLength("LINESTRING (0 0)", 0.0);
+
+        // Linestring with only one distinct point has length 0
+        assertLength("LINESTRING (0 0, 0 0, 0 0)", 0.0);
+
+        // ST_Length is equivalent to sums of ST_DISTANCE between points in the LineString
+        assertLength("LINESTRING (-71.05 42.36, -87.62 41.87, -122.41 37.77)", length);
+
+        // Linestring has same length as its reverse
+        assertLength("LINESTRING (-122.41 37.77, -87.62 41.87, -71.05 42.36)", length);
+
+        // Path north pole -> south pole -> north pole should be roughly the circumference of the Earth
+        assertLength("LINESTRING (0.0 90.0, 0.0 -90.0, 0.0 90.0)", 4.0075e7, 1e-2);
```

Comment:
Would it be possible to use the default tolerance here?

Reply:
The test fails with the default tolerance, but the difference in the output is only about 44 km. I can either change the estimate of earth's circumference and use the default tolerance or use the lower tolerance.

Reply:
Let's change the estimate.